The job **`dist-aarch64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/7302616593/job/19901604508) [(plain)](https://github.com/rust-lang-ci/rust/commit/5514de2c2ff6a17ccb9fe03981ce6c964bf5979e/checks/19901604508/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] panic_unwind test:false 0.070
[RUSTC-TIMING] std_detect test:false 0.105
[RUSTC-TIMING] hashbrown test:false 0.624
 Documenting std v0.0.0 (C:\a\rust\rust\library\std)
error[E0425]: cannot find value `SOMAXCONN` in crate `libc`
  --> library\std\src\os\unix\net\listener.rs:84:27
84 |                     libc::SOMAXCONN
   |                           ^^^^^^^^^ not found in `libc`


error[E0433]: failed to resolve: could not resolve path `libc::SOMAXCONN`
  --> library\std\src\os\unix\net\listener.rs:84:21
84 |                     libc::SOMAXCONN
84 |                     libc::SOMAXCONN
   |                     ^^^^^^^^^^^^^^^ could not resolve path `libc::SOMAXCONN`
   = note: this error was originally ignored because you are running `rustdoc`
   = note: try running again with `rustc` or `cargo check` and you may get a more detailed error

Some errors have detailed explanations: E0425, E0433.
Some errors have detailed explanations: E0425, E0433.
For more information about an error, try `rustc --explain E0425`.
error: could not document `std`

Caused by:
  process didn't exit successfully: `C:\a\rust\rust\build\bootstrap\debug\rustdoc --edition=2021 --crate-type dylib --crate-type rlib --crate-name std library\std\src\lib.rs --target aarch64-pc-windows-msvc -o C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\doc --cfg "feature=\"addr2line\"" --cfg "feature=\"backtrace\"" --cfg "feature=\"compiler-builtins-c\"" --cfg "feature=\"gimli-symbolize\"" --cfg "feature=\"miniz_oxide\"" --cfg "feature=\"object\"" --cfg "feature=\"panic_unwind\"" --cfg "feature=\"profiler\"" --cfg "feature=\"profiler_builtins\"" --cfg "feature=\"std_detect_dlsym_getauxval\"" --cfg "feature=\"std_detect_file_io\"" -Zunstable-options --check-cfg "cfg(feature, values(\"addr2line\", \"backtrace\", \"compiler-builtins-c\", \"compiler-builtins-mangled-names\", \"compiler-builtins-mem\", \"compiler-builtins-no-asm\", \"compiler-builtins-weak-intrinsics\", \"gimli-symbolize\", \"llvm-libunwind\", \"miniz_oxide\", \"object\", \"panic-unwind\", \"panic_immediate_abort\", \"panic_unwind\", \"profiler\", \"profiler_builtins\", \"std_detect_dlsym_getauxval\", \"std_detect_env_override\", \"std_detect_file_io\", \"system-llvm-libunwind\"))" --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat -C metadata=891dde679d70ac0b -L dependency=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps -L dependency=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\release\deps --extern alloc=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\liballoc-bba8830e940c1575.rmeta --extern priv:cfg_if=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libcfg_if-4b6fe407e7c31b0f.rmeta --extern priv:compiler_builtins=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libcompiler_builtins-a25b91a43e05c3e5.rmeta --extern core=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libcore-7ef9ece9e4ccbb66.rmeta --extern priv:hashbrown=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libhashbrown-103aea90a85074f5.rmeta --extern libc=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\liblibc-e18d78a2aa5e69be.rmeta --extern priv:panic_abort=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libpanic_abort-c9244d642eb4e0ba.rmeta --extern priv:panic_unwind=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libpanic_unwind-7ae2acf09575620b.rmeta --extern priv:profiler_builtins=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libprofiler_builtins-eb7097d97ef7d1de.rmeta --extern priv:rustc_demangle=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\librustc_demangle-294cbd20ea9356b6.rmeta --extern priv:std_detect=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libstd_detect-2beac49ef75ba228.rmeta --extern priv:unwind=C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2-std\aarch64-pc-windows-msvc\doc\aarch64-pc-windows-msvc\release\deps\libunwind-b1b3f08ebd661dab.rmeta -Z unstable-options -Csymbol-mangling-version=legacy -Zunstable-options --check-cfg=cfg(feature,values(any())) -Zunstable-options --check-cfg=cfg(bootstrap,values()) --check-cfg=cfg(stdarch_intel_sde,values()) --check-cfg=cfg(no_fp_fmt_parse,values()) --check-cfg=cfg(no_global_oom_handling,values()) --check-cfg=cfg(no_rc,values()) --check-cfg=cfg(no_sync,values()) --check-cfg=cfg(backtrace_in_libstd,values()) "--check-cfg=cfg(target_env,values(\"libnx\"))" "--check-cfg=cfg(target_arch,values(\"spirv\",\"nvptx\",\"xtensa\"))" -Dwarnings -Wrustdoc::invalid_codeblock_attributes --crate-version "1.77.0-nightly (5514de2c2 2023-12-22)" "-Zcrate-attr=doc(html_root_url=\"https://doc.rust-lang.org/nightly/\")" -Zcrate-attr=warn(rust_2018_idioms) -Z unstable-options --resource-suffix 1.77.0 --markdown-css rust.css --markdown-no-toc --index-page C:\a\rust\rust\src/doc/index.md --cfg backtrace_in_libstd` (exit code: 1)
[RUSTC-TIMING] std test:false 3.646
Build completed unsuccessfully in 0:25:12
  local time: Fri Dec 22 18:30:21 CUT 2023
  network time: Fri, 22 Dec 2023 18:30:21 GMT
```

</details>

--- a/library/std/src/os/unix/net/listener.rs
+++ b/library/std/src/os/unix/net/listener.rs
@@ -73,8 +73,18 @@ impl UnixListener {
         unsafe {
             let inner = Socket::new_raw(libc::AF_UNIX, libc::SOCK_STREAM)?;
             let (addr, len) = sockaddr_un(path.as_ref())?;
-            const backlog: libc::c_int =
-                if cfg!(any(target_os = "linux", target_os = "freebsd")) { -1 } else { 128 };
+            #[cfg(any(target_os = "windows", target_os = "redox"))]
+            const backlog: libc::c_int = 128;
+            #[cfg(any(target_os = "linux", target_os = "freebsd", target_os = "openbsd"))]
+            const backlog: libc::c_int = -1;
+            #[cfg(not(any(
+                target_os = "windows",
+                target_os = "redox",
+                target_os = "linux",
+                target_os = "freebsd",
+                target_os = "openbsd"
+            )))]
+            const backlog: libc::c_int = libc::SOMAXCONN;
 
             cvt(libc::bind(inner.as_inner().as_raw_fd(), &addr as *const _ as *const _, len as _))?;
             cvt(libc::listen(inner.as_inner().as_raw_fd(), backlog))?;

